File: frontend/src/app/dtos/order/cart.item.dto.ts
import { IsNumber } from 'class-validator';

export class CartItemDTO {
    @IsNumber()
    variant_id: number;

    @IsNumber()
    quantity: number;
    

    constructor(data: any) {
        this.variant_id = data.product_id;
        this.quantity = data.quantity;
    }
}

File: backend-microservices/order-service/src/main/java/com/project/orderservice/dtos/order/CartItemDTO.java
package com.project.orderservice.dtos.order;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class CartItemDTO {

    @JsonProperty("variant_id")
    private Long variantId;

    @JsonProperty("quantity")
    private Integer quantity;
}
